datafusion-cli/src/catalog.rs

```diff
@@ -236,13 +236,14 @@ mod tests {
     fn setup_context() -> (SessionContext, Arc<dyn SchemaProvider>) {
         let mut ctx = SessionContext::new();
         ctx.register_catalog_list(Arc::new(DynamicFileCatalog::new(
-            ctx.state().catalog_list(),
+            ctx.state().catalog_list().clone(),
```

Comment:
when I see `.clone()` Im now thinking what if we comment it the similar way as `.unwrap()` back in the day. Like say clone is cheap here because it is Arc::clone so only reference gets cloned.... thats just an idea, its too cumbersome to make it happen

Reply:
One thing we do in InfluxDB is use the pattern

```suggestion
            Arc::clone(ctx.state().catalog_list())
```
to make it explicit. 

There is a clippy lint https://rust-lang.github.io/rust-clippy/v0.0.212/#clone_on_ref_ptr we could turn on in datafusion to enable this 

Here is how it is done in influxdb:

https://github.com/influxdata/influxdb3_core/blob/0f5ecbd6b17f83f7ad4ba55699fc2cd3e151cf94/Cargo.toml#L117-L118

Reply:
This is a good idea, let's do it

Reply:
Filed https://github.com/apache/datafusion/issues/11143 to track